### PR TITLE
openvino 2024.3.0

### PR DESCRIPTION
**Pull request by artanokhov**
- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [ ] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/o/openvino.rb [hunk 1]
```diff
@@ -67,8 +67,8 @@ class Openvino < Formula
   end
 
   resource "onednn_cpu" do
-    url "https://github.com/openvinotoolkit/oneDNN/archive/373e65b660c0ba274631cf30c422f10606de1618.tar.gz"
-    sha256 "c205b81f9024952c742e765090556a18c9463fff245753a2afa42c344bd6379d"
+    url "https://github.com/openvinotoolkit/mlas/archive/d1bc25ec4660cddd87804fcf03b2411b5dfb2e94.tar.gz"
+    sha256 "13bee5b8522177f297e095e3eba5948c1a7ee7a816d19d5a59ce0f717f82cedc"
```
Comment (chenrui333) [inline Formula/o/openvino.rb:71]:
```suggestion
    sha256 "0a44fbfd4b13e8609d66ddac4b11a27c90c1074cde5244c91ad197901666004c"
```

**Comment by artanokhov** [comment]
@chenrui333 could you please take a look, I don't see any errors in the log, but
```
Error: 3 failed steps!
brew install --only-dependencies --verbose --formula --build-bottle ffmpeg@6
brew install --only-dependencies ./ffmpeg@6--6.1.1_3.arm64_monterey.bottle.tar.gz
brew install --only-dependencies --verbose --formula --build-bottle opencv
```

**Comment by artanokhov** [comment]
@chenrui333 @branchvincent 
Could you please look into the issues? Seems like OpenCV built failed because of timeout

```
Killing child processes...
Error: opencv: failed
::error::opencv: failed
An exception occurred within a child process:
  Timeout::Error: execution expired
/usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/3.3.4_1/lib/ruby/3.3.0/timeout.rb:43:in `rescue in handle_timeout'
/usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/3.3.4_1/lib/ruby/3.3.0/timeout.rb:40:in `handle_timeout'
/usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/3.3.4_1/lib/ruby/3.3.0/timeout.rb:195:in `timeout'
/usr/local/Homebrew/Library/Homebrew/test.rb:52:in `<main>'
```

**Comment by artanokhov** [comment]
Folks, could please anyone restart the pipelines?

**Comment by branchvincent** [comment]
Rerunning, I tried a re-run earlier with no luck. Maybe third time's the charm 🤞

**Comment by artanokhov** [comment]
Currently it also failing because of 3rd modules

```
==> Downloading https://ghcr.io/v2/homebrew/core/gnu-tar/blobs/sha256:b083b4ca16eea4b23615ce1b90b7e1a3ee52dd90cd5a4275567cb0ea55339ee4
  ==> Pouring gnu-tar--1.35.monterey.bottle.tar.gz
  ==> Fetching SBOM schema
  ==> Downloading https://spdx.github.io/spdx-3-model/model.jsonld
  curl: (22) The requested URL returned error: 404
  Warning: Failed to fetch SBOM schema, cannot perform SBOM validation!
  Warning: Failed to fetch SBOM schema, cannot perform SBOM validation!
  Error: Could not fetch JSON schema to validate SBOM!
  Error: Could not fetch JSON schema to validate SBOM!
  Error: Failure while executing; `/usr/local/bin/brew install --formula gnu-tar` exited with 1.
  Error: Failure while executing; `/usr/local/bin/brew install --formula gnu-tar` exited with 1.
```

I've checked - the url `https://spdx.github.io/spdx-3-model/model.jsonld` didn't exist

**Comment by artanokhov** [comment]
```
 ==> Downloading https://ghcr.io/v2/homebrew/core/opencv/manifests/4.10.0_2
  ==> Retrying download in 2s... (4 tries left)
  ==> Downloading https://ghcr.io/v2/homebrew/core/opencv/manifests/4.10.0_2
  ==> Retrying download in 4s... (3 tries left)
  ==> Downloading https://ghcr.io/v2/homebrew/core/opencv/manifests/4.10.0_2
  ==> Retrying download in 8s... (2 tries left)
  ==> Downloading https://ghcr.io/v2/homebrew/core/opencv/manifests/4.10.0_2
  ==> Retrying download in 16s... (1 try left)
  ==> Downloading https://ghcr.io/v2/homebrew/core/opencv/manifests/4.10.0_2
  Error: Failed to download resource "opencv_bottle_manifest"
  Couldn't find manifest matching bottle checksum.
  Error: Failed to download resource "opencv_bottle_manifest"
  Couldn't find manifest matching bottle checksum.
```
Still random issues not connected with our changes

Is there any possibility to re-start failed pipelines by ourselves?

**Comment by branchvincent** [comment]
Those logs are from the dep tests which fail because `opencv` wasn't built. The actual failure is the `opencv` [test has timed out](https://github.com/Homebrew/homebrew-core/actions/runs/10233567520/job/28516777060?pr=179077#step:3:109), which is now the third time so someone needs to investigate why that's happening before we run again

**Comment by artanokhov** [comment]
> Those logs are from the dep tests which fail because `opencv` wasn't built. The actual failure is the `opencv` [test has timed out](https://github.com/Homebrew/homebrew-core/actions/runs/10233567520/job/28516777060?pr=179077#step:3:109), which is now the third time so someone needs to investigate and fix why that's happening before we run again

If I understand correctly, opencv has a build, but the test of this simple version of printing via Python API hangs.
I don't think our changes could affect this, but maybe you have some suggestions how we can fix this?

Am I right in understanding that this is a timeout for the test or for the entire pipeline? If this is a timeout for the test - what is the default value now?

**Comment by artanokhov** [comment]
Noticed that there is already revision 2 for opencv (https://github.com/Homebrew/homebrew-core/blob/master/Formula/o/opencv.rb#L7), so I've bumped it to the 3

**Review by carlocab — APPROVED** [review]

**Comment by artanokhov** [comment]
Currently everything are failed because of `rtabmap` in deps build.
I'm definitely sure that this not affected by our changes

```
  make[2]: *** No rule to make target `/usr/local/Cellar/lz4/1.9.4/lib/liblz4.dylib', needed by `test'.  Stop.
  make[2]: *** Waiting for unfinished jobs....
  [ 50%] Building CXX object CMakeFiles/test.dir/test.cpp.o
  /Applications/Xcode_15.2.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang++ -DBOOST_ATOMIC_DYN_LINK -DBOOST_ATOMIC_NO_LIB -DBOOST_FILESYSTEM_DYN_LINK -DBOOST_FILESYSTEM_NO_LIB -DBOOST_IOSTREAMS_DYN_LINK -DBOOST_IOSTREAMS_NO_LIB -DBOOST_SERIALIZATION_DYN_LINK -DBOOST_SERIALIZATION_NO_LIB -DBOOST_SYSTEM_DYN_LINK -DBOOST_SYSTEM_NO_LIB -DEIGEN_HAS_CXX17_OVERALIGN=0 -DQT_CORE_LIB -DQT_GUI_LIB -DQT_NO_DEBUG -DQT_OPENGLWIDGETS_LIB -DQT_OPENGL_LIB -DQT_WIDGETS_LIB -Dkiss_fft_scalar=double -isystem /usr/local/include/eigen3 -isystem /usr/local/Cellar/rtabmap/0.21.4_4/include/rtabmap-0.21 -isystem /usr/local/opt/opencv/include/opencv4 -isystem /usr/local/include/pcl-1.14 -isystem /usr/local/include -isystem /usr/local/opt/libpcap/include -isystem /usr/local/Cellar/opencv/4.10.0_3/include/opencv4 -isystem /usr/local/include/vtk-9.3 -isystem /usr/local/include/vtk-9.3/vtknlohmannjson/include -isystem /usr/local/include/vtk-9.3/vtkfreetype/include -isystem /usr/local/lib/QtOpenGL.framework/Headers -iframework /usr/local/lib -isystem /usr/local/lib/QtCore.framework/Headers -isystem /usr/local/share/qt/mkspecs/macx-clang -isystem /usr/local/lib/QtGui.framework/Headers -isystem /usr/local/lib/QtWidgets.framework/Headers -isystem /usr/local/lib/QtOpenGLWidgets.framework/Headers -Os -w -pipe -march=westmere -mmacosx-version-min=13 -isysroot/Library/Developer/CommandLineTools/SDKs/MacOSX13.sdk -O3 -DNDEBUG -std=gnu++17 -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX13.sdk -msse4.2 -mfpmath=sse -fPIC -MD -MT CMakeFiles/test.dir/test.cpp.o -MF CMakeFiles/test.dir/test.cpp.o.d -o CMakeFiles/test.dir/test.cpp.o -c /tmp/rtabmap-test-20240819-53888-dcxkll/test.cpp
  make[1]: *** [CMakeFiles/test.dir/all] Error 2
  make: *** [all] Error 2
  
  Error: rtabmap: failed
  Error: rtabmap: failed
```

**Comment by SMillerDev** [comment]
> Could you please look into this issues?

Please don't tag volunteers, they'll spend time on it if they want.

**Comment by branchvincent** [comment]
`rtabmap` was fixed in https://github.com/Homebrew/homebrew-core/pull/181636, so need a rebase

**Review by carlocab — APPROVED** [review]
Thanks!

**Comment by artanokhov** [comment]
> > Could you please look into this issues?
> 
> Please don't tag volunteers, they'll spend time on it if they want.

Folks, I politely asked to look into the problems of the people who had already worked on this pull request.
But Ok.

**Pull request merged**